net: config: Allow initialization from application

jukkar opened the pull request:
If the CONFIG_NET_CONFIG_AUTO_INIT is not set, then let application
to do the initialization manually. In that case the app should call
net_config_init_app() when it is ready to setup the network.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: config: Allow initialization from application

If the CONFIG_NET_CONFIG_AUTO_INIT is not set, then let application
to do the initialization manually. In that case the app should call
net_config_init_app() when it is ready to setup the network.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
This is for USB networking where the initialization can be done when USB is ready. So application will need to do networking configuration when applicable.

emob-nordic commented:
`CONFIG_NET_CONFIG_INIT_TIMEOUT` is available only if `CONFIG_NET_CONFIG_AUTO_INIT=y` which leads to build error with USB overlay. Could this not depend on `*_AUTO_INIT`?

eobalski reviewed (changes requested):
```diff
diff --git a/subsys/net/lib/config/Kconfig b/subsys/net/lib/config/Kconfig
index 9220aa011e..0669f10152 100644
--- a/subsys/net/lib/config/Kconfig
+++ b/subsys/net/lib/config/Kconfig
@@ -19,7 +19,6 @@ config NET_CONFIG_INIT_PRIO
 config NET_CONFIG_INIT_TIMEOUT
 	int "How long to wait for networking to be ready and available"
 	default 30
-	depends on NET_CONFIG_AUTO_INIT
 	help
 	  The value is in seconds. If for example IPv4 address from DHCPv4 is not
 	  received within this limit, then the net_config_init() call will fail
```

jukkar commented:
> Could this not depend on `*_AUTO_INIT`?

Yes, your change make sense. Can you prepare a commit so I can add it to this PR?

Commit:
net: Do not depend timeout on AUTO_INIT.

CONFIG_NET_CONFIG_INIT_TIMEOUT should not only be available if
CONFIG_NET_CONFIG_AUTO_INIT=y. If *_AUTO_INIT=n then application
may wish to initialize net manually and still use configured
timeout.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

emob-nordic commented:
> > Could this not depend on `*_AUTO_INIT`?
> 
> Yes, your change make sense. Can you prepare a commit so I can add it to this PR?

Yea, np.

emob-nordic commented:
You can cherry-pick form [this](https://github.com/emob-nordic/zephyr/commits/net_allow_init_from_app). I think 1st commit based on Your branch should be squashed to your commit.
